tests/integration_tests.py

```diff
@@ -647,6 +647,350 @@ def test_will_not_output_empty_state(self):
+
+        self.messages_until_error = 1
+        target_stitch.OUR_SESSION = FakeSession(mock_in_order_all_200)
+
+        self.queue.append(json.dumps({"type": "RECORD", "stream": "chicken_stream", "record": {"id": 1}}))
+        self.queue.append(json.dumps({"type": "STATE",  "value": {"bookmarks": {"chicken_stream": {"id": 1}}}}))
+
+        self.queue.append(json.dumps({"type": "RECORD", "stream": "zebra_stream", "record": {"id": 1}}))
+        self.queue.append(json.dumps({"type": "STATE",  "value": {"bookmarks": {"chicken_stream": {"id": 1},
+                                                                                "zebra_stream": {"id": 1}}}}))
+
+        self.queue.append(json.dumps({"type": "RECORD", "stream": "dog_stream", "record": {"id": 1}}))
+        self.queue.append(json.dumps({"type": "STATE",  "value": {"bookmarks": {"chicken_stream": {"id": 1},
+                                                                                "zebra_stream": {"id": 1},
+                                                                                "dog_stream": {"id": 1}}}}))
+
+        self.queue.append(json.dumps({"type": "RECORD", "stream": "chicken_stream", "record": {"id": 2}}))
+        self.queue.append(json.dumps({"type": "STATE",  "value": {"bookmarks": {"chicken_stream": {"id": 2},
+                                                                                "zebra_stream": {"id": 1},
+                                                                                "dog_stream": {"id": 1}}}}))
+
+        self.queue.append(json.dumps({"type": "RECORD", "stream": "zebra_stream", "record": {"id": 2}}))
+        self.queue.append(json.dumps({"type": "STATE",  "value": {"bookmarks": {"chicken_stream": {"id": 2},
+                                                                                "zebra_stream": {"id": 2},
+                                                                                "dog_stream": {"id": 1}}}}))
+
+        self.queue.append(json.dumps({"type": "RECORD", "stream": "dog_stream", "record": {"id": 2}}))
+        self.queue.append(json.dumps({"type": "STATE",  "value": {"bookmarks": {"chicken_stream": {"id": 2},
+                                                                                "zebra_stream": {"id": 2},
+                                                                                "dog_stream": {"id": 2}}}}))
+
+
+        self.target_stitch.consume(self.queue)
+
+        try:
+            finish_requests()
+        except:
+            pass
+
+        # There should only be messages for the 2 streams because the
+        # third one should fail due to the mocking code
+        expected_messages = [[{'action': 'upsert', 'data': {'id': 1}},
+                              {'action': 'upsert', 'data': {'id': 2}}],
+                             [{'action': 'upsert', 'data': {'id': 1}},
+                             {'action': 'upsert', 'data': {'id': 2}}]]
+
+        expected_state = ''
```

Comment:
Same here

Reply:
No. Check out the `mock_post_coroutine()`. In this test the first batch is failing to send.